cloudinit/url_helper.py

```diff
@@ -639,61 +724,60 @@ def dual_stack(
     return (returned_address, return_result)
 
 
+class HandledResponse(NamedTuple):
```

Comment:
I don't love this honestly, but it provides a more structured way of return stuff from the nested functions in `wait_for_url()`